projects/hslayers/src/components/add-data/file/shp/add-data-file-shp.service.ts

```diff
@@ -72,15 +96,25 @@ export class HsAddDataFileShpService {
           {withCredentials: true}
         )
         .toPromise()
-        .then((data: any) => {
-          if (data && data.length > 0) {
-            resolve(data);
+        .then(async (data: any) => {
+          //CHECK IF OK not auth etc.
+          if (data && data.length > 0) {  
+            if (this.asyncLoading) {
+              const promise = await this.HsLaymanService.asyncUpload(
+                files_to_async_upload,
+                data,
+                endpoint
+              );
+              resolve(promise);
+            } else {
+              resolve(data);
+            }
           } else {
             reject(data);
           }
         })
         .catch((err) => {
-          this.hsLog.error(err);
+          console.error(err);
```

Comment:
hsLog is our wrapper around console object, why replacing it?

Reply:
Without hs service the call stack tree is 'clickable', was easier for me to debug and I just forgot to revert

Reply:
Will have to make hsLog also clickable.

Reply:
I can see. I use the console.log for debugging as well. I only switch to hsLog when I feel that the log could be useful for debugging possible runtime errors on production. Yes, clickable hsLog would a nice thing to have 🙂